next.config.mjs

```diff
@@ -397,173 +571,459 @@ const nonPermanentRedirects = [
-  ["/self-hosting/infrastructure/clickhouse", "/self-hosting/deployment/infrastructure/clickhouse"],
-  ["/self-hosting/infrastructure/containers", "/self-hosting/deployment/infrastructure/containers"],
-  ["/self-hosting/infrastructure/llm-api", "/self-hosting/deployment/infrastructure/llm-api"],
-  ["/self-hosting/infrastructure/postgres", "/self-hosting/deployment/infrastructure/postgres"],
+  [
+    "/self-hosting/infrastructure/blobstorage",
+    "/self-hosting/deployment/infrastructure/blobstorage",
+  ],
+  [
+    "/self-hosting/infrastructure/cache",
+    "/self-hosting/deployment/infrastructure/cache",
+  ],
+  [
+    "/self-hosting/infrastructure/clickhouse",
+    "/self-hosting/deployment/infrastructure/clickhouse",
+  ],
+  [
+    "/self-hosting/infrastructure/containers",
+    "/self-hosting/deployment/infrastructure/containers",
+  ],
+  [
+    "/self-hosting/infrastructure/llm-api",
+    "/self-hosting/deployment/infrastructure/llm-api",
+  ],
+  [
+    "/self-hosting/infrastructure/postgres",
+    "/self-hosting/deployment/infrastructure/postgres",
+  ],
   ["/self-hosting/kubernetes-helm", "/self-hosting/deployment/kubernetes-helm"],
   ["/self-hosting/railway", "/self-hosting/deployment/railway"],
-  ["/self-hosting/authentication-and-sso", "/self-hosting/security/authentication-and-sso"],
-  ["/self-hosting/deployment-strategies", "/self-hosting/security/deployment-strategies"],
+  [
+    "/self-hosting/authentication-and-sso",
+    "/self-hosting/security/authentication-and-sso",
+  ],
+  [
+    "/self-hosting/deployment-strategies",
+    "/self-hosting/security/deployment-strategies",
+  ],
   ["/self-hosting/networking", "/self-hosting/security/networking"],
   ["/self-hosting/troubleshooting", "/self-hosting/troubleshooting-and-faq"],
-  ["/self-hosting/background-migrations", "/self-hosting/upgrade/background-migrations"],
-  ["/self-hosting/upgrade-guides/upgrade-v1-to-v2", "/self-hosting/upgrade/upgrade-guides/upgrade-v1-to-v2"],
-  ["/self-hosting/upgrade-guides/upgrade-v2-to-v3", "/self-hosting/upgrade/upgrade-guides/upgrade-v2-to-v3"],
+  [
+    "/self-hosting/background-migrations",
+    "/self-hosting/upgrade/background-migrations",
+  ],
+  [
+    "/self-hosting/upgrade-guides/upgrade-v1-to-v2",
+    "/self-hosting/upgrade/upgrade-guides/upgrade-v1-to-v2",
+  ],
+  [
+    "/self-hosting/upgrade-guides/upgrade-v2-to-v3",
+    "/self-hosting/upgrade/upgrade-guides/upgrade-v2-to-v3",
+  ],
   ["/self-hosting/versioning", "/self-hosting/upgrade/versioning"],
+
+  // Redirect TypeScript SDK v3 docs to new docs
+  [
+    "/docs/observability/sdk/typescript/guide",
+    "/docs/observability/sdk/typescript/overview",
+  ],
+  [
+    "/docs/observability/sdk/typescript/guide-web",
+    "/docs/observability/sdk/typescript/overview",
+  ],
+  [
+    "/docs/observability/sdk/typescript/example-notebook",
+    "/docs/observability/sdk/typescript/overview",
+  ],
+  [
+    "/guides/cookbook/js_integration_llamaindex",
```

Comment:
does not sound like the best redurect if we still have an example notebook in cookbook folder, should redirect to /guides